ls1intum/Artemis
# `Exam mode`: Persist live statistics actions

### Checklist
#### General


#### Server

### Motivation and Context



In order to store the performed actions, we need to provide a database schema. In addition, we want to store the actions during the night.

### Description


- Added some Entities
- Updated the save logic (we persist the actions at 3 am) 

### Steps for Testing

Prerequisites:
- 1 Instructor
- One/Multiple Student
- 1 Exam

1. Log in to Artemis
2. Create an Exam with Exam Live Statistics enabled
3. Participate with the student
4. Verify that the sections work as expected
5. Verify that the actions are stored in the database

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| ExamMonitoringScheduleService.java | 80% | 91% |

### Screenshots



![PersistentDataManagement](https://user-images.githubusercontent.com/60005702/181368919-3d28ef28-a2cf-47d2-8dca-4fca136cfa4b.png)
